journey-builder-triggers/template.ts

```diff
@@ -0,0 +1,161 @@
+                    break;
+                case "EventAction":
+                    trigger = context.trigger as ActionTrigger;
+                    result["Trigger_Action"] = trigger.action;                     
+                    break;
+                default:
+                    trigger = context.trigger as CatalogTrigger;
+                    const items: any[] = [];
+
+                    if(trigger.itemIdsByType){
+                        Object.keys(trigger.itemIdsByType).forEach(catalogObject => {
+                            items.push(...JourneyBuilderTriggersUtils.getFlatItems(context, catalogObject, trigger.itemIdsByType[catalogObject]));
+                        });
+                    }
+
+                    try {
+                        result["Trigger_Catalog_Items"] = JSON.stringify(items);
+                    } catch (ex) {}
+                    break;
+            }
+        }
+
+        return result;
+    }
+}
+
+export class JourneyBuilderTriggersPayload {
+    @title(" ")
+    @header("User Attributes:")
+    @headerSubtitle("After you select user attributes, create a field for each user attribute in the event data extension in Marketing Cloud. Data extension field names must match user attribute names.")
+    @lookupOptions(() => new UserAttributeLookup())
+    attributes: UserAttributeReference[];
+
+    @title(" ")
+    @header("Recommendations:")
+    @headerSubtitle("After you select a recipe, create a Recommendations field in the event data extension in Marketing Cloud. Doing so allows you to add a Recommendations block to an email in Marketing Cloud.")
+    @lookupOptions(() => new RecipeReferenceLookup(JourneyBuilderTriggersUtils.defaultCatalogObject))
+    recipe: RecipeReference;
+
+    @title("Include Additional Recommendations")
+    @subtitle("After you select a recipe, create an Additional_Recommendations field in the event data extension in Marketing Cloud. Doing so allows you to add an Additional Recommendations block to an email in Marketing Cloud.")
+    includeAdditionalRecommendations: boolean = false;
+
+    @title(" ")
+    @shownIf(this, self => self.includeAdditionalRecommendations)
+    @lookupOptions(() => new RecipeReferenceLookup(JourneyBuilderTriggersUtils.defaultCatalogObject))
+    additionalRecipe: RecipeReference;
+
+    @title("Include User Segments")
+    @subtitle("After you select segments, create an Additional_Segment field in the event data extension in Marketing Cloud.")
+    includeSegments: boolean = false;
+
+    @title(" ")
+    @shownIf(this, self => self.includeSegments)
+    @lookupOptions(() => new UserSegmentLookup())
+    segments: UserSegmentReference[];
+
+    getPayload(context: CampaignComponentContext){
+        const triggerContext = JourneyBuilderTriggersUtils.getTriggerContext(context);
+        
+        const result = {
+            "Campaign": context.campaignId,
+            "Experience": context.experienceId,
+            ...triggerContext
+        };
```

Comment:
minor: would `triggerContext` ever contain the keys `"Campaign"` and/or `"Experience"`? And if so and if we want the keys from `context` to take preference, then this should be:

```suggestion
        const result = {
            ...triggerContext,
            "Campaign": context.campaignId,
            "Experience": context.experienceId,
        };
```

Reply:
This shouldn't be the case. `triggerContext` is built through `getTriggerContext` which statically builds an object of `Trigger_Type` and optionally `Trigger_Segment`, `Trigger_Action` or `Trigger_Catalog_Items` based on the trigger type.